cas-server-support-osf/src/main/java/io/cos/cas/web/flow/OpenScienceFrameworkLoginHandler.java

```diff
@@ -56,22 +56,29 @@
         private boolean orcidRedirect;
 
         /**
-         * Construct an instance of `OpenScienceFrameworkLoginContext` with the given settings.
+         * Construct an instance with the service URL, the institution login and ORCiD redirect flags.
          *
          * @param serviceUrl the service URL
          * @param institutionLogin the flag for institution login
+         * @param orcidRedirect the flag for ORCiD redirect
          */
-        private OpenScienceFrameworkLoginContext(final String serviceUrl, final boolean institutionLogin) {
+        private OpenScienceFrameworkLoginContext(
+                final String serviceUrl,
+                final boolean institutionLogin,
+                final boolean orcidRedirect
+        ) {
             this.serviceUrl = serviceUrl;
             this.handleErrorName = null;
             this.institutionLogin = institutionLogin;
+            this.orcidRedirect = orcidRedirect;
         }
 
+        // Must be public to be accessible in the JSP page
         public String getServiceUrl() {
             return serviceUrl;
         }
 
-        public void setServiceUrl(final String serviceUrl) {
+        void setServiceUrl(final String serviceUrl) {
```

Comment:
Other getters, setters and helpers such as `setServiceUrl()` can be "package-private".